### ISSUE TITLE
DeskPi Pro V2 - temperature is read, fan mode is adjusted, fan doesn't spin at all.

### ISSUE THREAD

**Issue by Jimmypl007** [adamoutler/HassOSDeskPiPro#6]
I have installed [HassOS Serial Configurator] and it seems to be working, I can plug a USB in the front two ports and it seems to be recognized in the HomeAssistant (supervisor) system log. 
Using `/dev/bus/usb/003/004` for the left one and `/dev/bus/usb/003/005` for the one on the right.
(_as I mentioned it seems to be working, I never did plug in a USB before using the configurator so maybe it always worked and I miss understand exactly what the configurator is meant to do.._)
they also had these below which seems correct to what the configurator said it would do.

_left front port._
```
/dev/bus/usb/003/004
/dev/bsg/1:0:0:0 - None
/dev/sdb - /dev/disk/by-id/usb-SanDisk_Cruzer_Edge_4C530599931113109125-0:0
/dev/sdb1 - /dev/disk/by-id/usb-SanDisk_Cruzer_Edge_4C530599931113109125-0:0-part1
```
_right front port._
```
/dev/bus/usb/003/005
/dev/bsg/1:0:0:0 - None
/dev/sdb - /dev/disk/by-id/usb-SanDisk_Cruzer_Edge_4C530599931113109125-0:0
/dev/sdb1 - /dev/disk/by-id/usb-SanDisk_Cruzer_Edge_4C530599931113109125-0:0-part1
```

I don't know if it helps but I saw you mentioned it. [here](https://community.home-assistant.io/t/add-on-hassos-serial-configurator/264169/2) that if I use a Zigbee dongle in `/dev/ttyUSB0` i would have to reconfigure something.

When HA had recovered from the Hard shutdown and restart, I noticed zigbee2mqtt failed to launch and had a bunch of errors (_I figured this was odd as I used the specific device address and not the port_), so I went and pulled the dongle out and put it back in to get the new address thinking maybe it had changed. This is what popped up `/dev/ttyUSB0 - /dev/serial/by-id/usb-1a86_USB_Serial-if00-port0` the second part being my specific dongle. As you can see it has not changed from `ttyUSB0` to `ttyUSB1` or `ttyUSB2`.

This addon (HassOSDeskPiPro) specifies in 'configuration' the serial device* `/dev/ttyUSB0`. Is this meant to point to the fan? if so I can see why the fan is not powering on.  should I try to figure out what the fan is connected to ??

(below is a printout, seems to be just like everyone else I have seen, I saw you mention in another issue that `nc: bind: Address in use` is not related to the issue) if it looks cool, it's because I have AC cranked in that room.
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
nc: bind: Address in use
Current Temperature 46 °C
Quiet Level 3 - Fan 50% (Medium)
Current Temperature 45 °C
Quiet Level 2 - Fan 20% (Low)
Current Temperature 46 °C
Quiet Level 3 - Fan 50% (Medium)
Current Temperature 46 °C
Current Temperature 46 °C
Current Temperature 45 °C
Quiet Level 2 - Fan 20% (Low)
Current Temperature 45 °C
Current Temperature 45 °C
Current Temperature 46 °C
Quiet Level 3 - Fan 50% (Medium)
Current Temperature 45 °C
Quiet Level 2 - Fan 20% (Low)
Current Temperature 45 °C
Current Temperature 45 °C
Current Temperature 45 °C
Current Temperature 45 °C
```

any suggestions would be great...

**Comment by Jimmypl007** [comment]
I am getting more and more convinced that the fan is meant to be on USB0, at the moment it is not. it would make sense as to why the configurator was meant to move the ports up by 1, my Zigbee dongle is still on usb0. I just took the m.2 out of the pie to double-check and this (below) is listed at the bottom of `config.txt` , so it seems that I ran the configurator correctly.
```
[all]
#dtoverlay=vc4-fkms-v3d
#max_framebuffers=2
dtoverlay=dwc2,dr_mode=host
```
_I have to be missing something, is there something that is meant to expose the fan to usb0?
if it's not obvious yet I am a newb at all this..._

Update:

I removed all USB devices except the little USB to USB adapter that comes with the deskpi to connect the m.2 to the rasp pi. ( i wouldn't be able to boot without it) now, when running `ls /dev` i get only `ttyUSB0` but I am guessing (if the serial device in the configuration menu has to point at the fan) that `ttyUSB0` now refers to the adapter dongle. as the fan is still not spinning.

tomorrow I plan on starting a new home assistant on an SD card to rule out the dongle being a USB and confirming whether or not the `dtoverlay=dwc2,dr_mode=host` has done anything.. hopefully, I can figure stuff out, in the meantime if you see this and can suggest anything please do, and I apologize if this is spammy.

**Comment by Jimmypl007** [comment]
Today I booted up an SD card and with nothing on it, I followed the tutorial listed in the [configurator addon comments](https://community.home-assistant.io/t/add-on-hassos-serial-configurator/264169/6) and to my amazement, the fan started spinning (although a little noisy) so i went back to the m.2 i followed the tutorial again and checked everything (`ls /dev`) along the way, it was all the same except it just didnt turn the fan on.

the USB0 is not the usb adapter as mentioned above. im sure anyone that knows this sort of stuff could have said that. its actually given some other name. so the fact ttyUSB0 had appeared means it should just have worked..

my next step is now to back everything up on my HA and try a fresh install on the m.2 this way it starts identical to the SD card

my next step IF that doesnt work (and man i will be confused if it doesn't..).. will be to make a backup of HA from the working SD card version and load it up on the m.2 if possible..

**Comment by adamoutler** [comment]
The `/dev/ttyUSB0` is the default.  Add-ons can't detect system settings they must be given by the user.  What was your device?

**Comment by Jimmypl007** [comment]
I didn't think I would get an answer and I have literally come back just now to say what was going on.
Thanks for the reply though.

It turns out when I restart my system (full power off/ plug-out) with my Zigbee dongle plugged in, it will always take ttyUSB0. _not sure if this is expected behavior for HA but it obviously conflicts with this addon and the way we are required to do a "hard shut down" and start for the configurator._

Anyway, after I did a clean install with my m.2 as I mentioned above, I found i got the addon working completely, I was very happy, then when I shut down and put the case back together and restarted (with the Zigbee dongle plugged in) I discovered it took USB0 and the fan no longer worked I figured if I rebooted without the dongle it would work, and it did. I can just plug the Zigbee dongle in after a fresh boot and it works although this is not preferable.

To avoid removing the Zigbee dongle every full power off, (and in case of power outs) could I set the deskPi addon configuration to usb1? obviously, I will try this tomorrow but it's late now.

thanks for any advice'

EDIT: I notice I keep switching the words "dongle" and "adapter" I have edited it all to dongle now.

**Comment by adamoutler** [comment]
Yes.  You could just set it to `/dev/ttyUSB1`.   Alternatively, and most appropriately but most difficult, you could set up a [Udev](https://opensource.com/article/18/11/udev) rule to force naming of the devices.

**Comment by adamoutler** [comment]
Alternatively, I've never tried it, but the USB port in the front is initialized after the stock USB.  It may be possible to reverse the order by plugging the dongle into the front USB.

**Comment by Jimmypl007** [comment]
> Yes.  You could just set it to `/dev/ttyUSB1`.   Alternatively, and most appropriately but most difficult, you should set up a [Udev](https://opensource.com/article/18/11/udev) rule to force naming of the devices.  
> 

Hmm i tried setting it to usb2 and boy did it not like that. The fan did turn on with max speed which was its previous state before shutting down. But then it wouldn't update when i re adjusted the config. It was no longer getting commands from the addon.

Also despite the zigbee dongle being usb0 and zigbee2mqtt looking at the dongle by name it refused to be able to find it..

Clearly there is much more going on than i know.. 

Anyway reverting back to the fan as usb0 and booting, then plugging in the zigbee dongle as usb1 seemed to fix everything for now. With of course the con of having to do that and boot zigbee2mqtt manually if i do a power down and out or there is a power out.

I will have to look into this udev i saw you mention it in an older closed issue.. i wouldnt know where to begin to be honest but hey i got this far..

Again thanks for the response
EDIT: i see you gave a link, much appreciated. I will take a look when i can.

**Comment by Jimmypl007** [comment]
> Alternatively, I've never tried it, but the USB port in the front is initialized after the stock USB.  It may be possible to reverse the order by plugging the dongle into the front USB. 

Will give this a shot tommorow too.
Thanks again.

**Comment by Jimmypl007** [comment]
> Alternatively, I've never tried it, but the USB port in the front is initialized after the stock USB. It may be possible to reverse the order by plugging the dongle into the front USB.

This didnt work either, the adapter comes up as USB0 as usual.

**However! it looks like I have found a fix.**

I thought to myself, with Zigbee2MQTT i can specify the device with `/dev/serial/by-id/usb-1a86_USB_Serial-if00-port0`
and figured perhaps I could point the desk pi Addon straight at the fan.

I noticed the device paths similarity to the command `ls /dev` and figured that's what the command is showing me.
So I typed in `ls /dev/serial/by-id/` in the SSH Terminal and it listed one item (at the time I didn't have the Zigbee dongle plugged in.) However, I noticed the item in the list was exactly `usb-1a86_USB_Serial-if00-port0` (_the same as what I thought was specifically my Zigbee dongle, I guess I was wrong_).
Sure enough, after plugging in the Zigbee dongle and running the command again, the list did not change. it continued to only show one item.

I figured it must be showing somewhere else so I went back one level to `ls /dev/serial/` and after pressing enter was given another option along with by ID , `by-path` typing that command in `ls /dev/serial/by-path/` showed two devices 
`platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.4:1.0-port0`
`platform-fe980000.usb-usb-0:1.4:1.0-port0`
and while they seemed obvious I made sure I knew which each one was by unplugging the Zigbee dongle.
I applied the `fd500000` device (the Zigbee dongle) to my Zigbee2MQTT configuration and then applied the `fe980000` device to your Deskpi Addon using the path structure - `/dev/serial/by-path/Then_the_platform_device`

A complete shutdown, power out, and start came up with everything working.

_tbh I don't know how this is any different from when I set the fan config to USB2 when the fan was USB2. obviously, if this does what I think it is doing then I won't have to reconfigure the fan every time I add another USB device._
